### PR TITLE
allow for custom filtering of doctests

### PR DESCRIPTION
**Pull request by KristofferC**
Replaces https://github.com/JuliaDocs/Documenter.jl/pull/540 and perhaps https://github.com/JuliaDocs/Documenter.jl/pull/549.

### CODE REVIEW

**Comment by KristofferC** [comment]
This could easily be extended to arbitrary functions and not only regular expressions but I kept it simple to start with.

File: docs/src/man/doctests.md [hunk 1]
```diff
@@ -191,6 +191,35 @@ help avoid unintentional definitions in following doctest blocks.
...
+## Filtering Doctests
+
+A part of the output of a doctest might be non-deterministic, e.g. pointer addresses and timings.
+It is therefore possible to filter a doctest so that the deterministic part can still be tested.
+A filter takes the form of a regular expression.
+Each match in the expected output and the actual output is removed before the outputs are compared. 
+Filters are added globally i.e. applied to all doctests in the documentation by giving a list of regular expressions to
+`makedocs` with the keyword `doctestfilters`.
+For more fine grained control, this list can also be assigned inside a `@meta` block by assigning to the variable `DocTestFilters`.
+The global filters and those defined in the `@meta` block are concatinated.
```
Comment (bicycle1885) [inline docs/src/man/doctests.md]: concatenated?

File: docs/src/man/doctests.md [hunk 2]
```diff
@@ -191,6 +191,35 @@ help avoid unintentional definitions in following doctest blocks.
     The `DocTestSetup` value is **re-evaluated** at the start of *each* doctest block
     and no state is shared between any code blocks.
 
+## Filtering Doctests
+
+A part of the output of a doctest might be non-deterministic, e.g. pointer addresses and timings.
+It is therefore possible to filter a doctest so that the deterministic part can still be tested.
+A filter takes the form of a regular expression.
```
Comment (mortenpi) [inline docs/src/man/doctests.md]: I would have this sentence start a new paragraph...

File: docs/src/man/doctests.md [hunk 3]
```diff
@@ -191,6 +191,35 @@ help avoid unintentional definitions in following doctest blocks.
...
+Each match in the expected output and the actual output is removed before the outputs are compared. 
+Filters are added globally i.e. applied to all doctests in the documentation by giving a list of regular expressions to
```
Comment (mortenpi) [inline docs/src/man/doctests.md]: ... and this one too. Splitting this paragraph up a bit would improve readability a bit in my opinion.

**Comment by KristofferC** [comment]
Rebased on top of #574, also added some doctests. And fixed the style thing in the docs.

**Comment by KristofferC** [comment]
Rebased

**Comment by mortenpi** [comment]
Awesome. Thanks a lot @KristofferC for all the PRs, much appreciated! Will tag shortly.

**Pull request merged**